konfera/views.py

```diff
@@ -1,3 +1,14 @@
 from django.shortcuts import render
+from konfera.models import Event, Sponsor
 
-# Create your views here.
+
+def sponsor_list_view(request):
+    events = Event.objects.all()
+    sponsors = Sponsor.objects.all()
```

Comment:
This is not very effective, you are doing select for all sponsors and all events. This would be much more effective since if we have million event and 10 sponsors per event, you will do two DB queries that will result in 11 million results, which you will later order and cycle through.

May be I wasn't clear enough in the issue, it should be all sponsors per event. So in url you define event, and in the view [.get](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#get) it and than just [.filter](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#filter) the sponsors for that event. This way you will do DB queries that will result in 11 results, which you will later order and cycle through.

And if you check out [.select_related](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#select-related) Django will construct more complicated SQL query that will follow foreign keys and you will have only DB query and in the template you will have only one for cycle.